docs/guides/javascript.md

```diff
@@ -199,3 +202,28 @@ const spectral = new Spectral({ resolver: customFileResolver });
 The custom resolver we've just created will resolve all remote file refs relatively to the current working directory.
 
 More on that can be found in the [json-ref-resolver repo](https://github.com/stoplightio/json-ref-resolver).
+
+### Using custom de-duplication strategy
+
+By default, Spectral will de-duplicate results based on the result code and document location. You can customize this
```

Comment:
How about using deduplicateResults function I implemented back in a day and nulltoken vastly improved recently? I think we just don't call it in a run method. We just call it in the CLI mode.

Reply:
We could - however the fingerprint approach is more elegant IMHO, and easy to customize - does deduplicateResults cover things the default compute fingerprint func in this PR does not?

Reply:
@marbemac internally `deduplicateResults` relies on a sorting comparer. As such, the deduplicated results are also... sorted ;-) This allows the linter to always provide a predictable and stable output in the test harness and cli output.

However, I tend to agree with @P0lip that having two "somewhat-but-not-completely-similar" functions might not be ideal. How would you feel about making deduplicateResults leverage this fingerprint based approach and then eventually sort the resulting array of results?